__tests__/unit/fact.test.js

```diff
@@ -0,0 +1,415 @@
+      },
+    });
+    expect(sentence).toBe(
+      "Sweyn I establishes Danish control over part of Norway"
+    );
+  });
+
+  test("When wantFragment is undefined, returns full sentence", function () {
+    const sentence = getSentence({
+      wantFragment: undefined,
+      number: 1000,
+      type: "year",
+      data: {
+        text: "Sweyn I establishes Danish control over part of Norway",
+      },
+    });
+    expect(sentence).toBe(
+      "1000 is the year that Sweyn I establishes Danish control over part of Norway."
+    );
+  });
+
+  test("When data has key of date, returns full sentence with date in the end", function () {
+    const sentence = getSentence({
+      wantFragment: undefined,
+      number: 1000,
+      type: "year",
+      data: {
+        date: "December 25",
+        text:
+          "Stephen I becomes King of Hungary, which is established as a Christian kingdom",
+      },
+    });
+    expect(sentence).toBe(
+      "1000 is the year that Stephen I becomes King of Hungary, which is established as a Christian kingdom on December 25th."
+    );
+  });
+
+  test("Return sentence when year is invalid", function () {
+    const sentence = getSentence({
+      wantFragment: undefined,
+      type: "year",
+      number: 10000000000,
+      data: {
+        text: "nothing remarkable happened",
+      },
+    });
+    expect(sentence).toBe(
+      "10000000000 is the year that nothing remarkable happened."
+    );
+  });
+});
+
+describe("getSentence() for types: trivia and math", () => {
+  // trivia and math share the same prefix
+  test("When wantFragment is true, returns partial text only", function () {
+    const sentence = getSentence({
+      wantFragment: { fragment: true },
+      number: 88,
+      type: "trivia",
+      data: {
+        text: "the number of keys on a piano (36 black and 52 white)",
+      },
+    });
+    expect(sentence).toBe(
+      "the number of keys on a piano (36 black and 52 white)"
+    );
+  });
+
+  test("When wantFragment is undefined, returns full sentence", function () {
+    const sentence = getSentence({
+      wantFragment: undefined,
+      number: 88,
+      type: "trivia",
+      data: {
+        text: "the number of keys on a piano (36 black and 52 white)",
+      },
+    });
+    expect(sentence).toBe(
+      "88 is the number of keys on a piano (36 black and 52 white)."
+    );
+  });
+
+  test("When trivia number is invalid, returns sentence", function () {
+    const sentence = getSentence({
+      wantFragment: undefined,
+      number: 88888,
+      type: "trivia",
+      data: {
+        text: "an unremarkable number",
+      },
+    });
+    expect(sentence).toBe("88888 is an unremarkable number.");
+  });
+});
+
+describe("getSentence() for type 'date' ", function () {
+  test("When wantFragment is undefined, returns full sentence", function () {
+    const sentence = getSentence({
+      wantFragment: undefined,
+      number: 1 / 1,
```

Comment:
why is this `1 / 1` instead of just `1`? if there's a reason, can you add a comment giving it? if there's no reason, let's just use `1`